### ISSUE TITLE
Screen goes blank in debug mode

### ISSUE THREAD

**Issue by m-waqas88** [ethereum/remix-project#3324]
With environment Injected Provider - Metamask, the screen goes blank when debug button is clicked. I am trying to debug `createToken` function.

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.4;

import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
import "@openzeppelin/contracts/utils/Counters.sol";

contract NFT is ERC721URIStorage {
    using Counters for Counters.Counter;
    Counters.Counter private _tokenIds;
    address contractAddress;
    
    constructor(address marketplaceAddress) ERC721("Metaverse Tokens","METT") {
        contractAddress=marketplaceAddress;
    
    }

   function createToken(string memory tokenURI) public returns(uint){
    
       _tokenIds.increment();
       uint256 newItemId=_tokenIds.current();
       _mint(msg.sender,newItemId); 
       _setTokenURI(newItemId,tokenURI);

       setApprovalForAll(contractAddress,true);
       return newItemId;
       
   } 
}

**Comment by Aniket-Engg** [comment]
Hello @m-waqas88 , Can you inform about the network set in Metamask? Also can you check if it is fixed on remix-beta.ethereum.org ?

**Comment by m-waqas88** [comment]
@Aniket-Engg 

The network set in Metamask is HardHat localhost:8545.

When I tried this on remix-beta then it throws below error:
`The method "debug_traceTransaction" does not exist / is not available.`

I also tried setting the network as Goerli Testnet. In this case, I am facing the same error as I mentioned in issue https://github.com/ethereum/remix-project/issues/2529

**Comment by m-waqas88** [comment]
@Aniket-Engg I get your point. So #2529 will be resolved in the next release, great.

Regarding this issue, in remix-beta.ethereum.org it shows me the error that `The method "debug_traceTransaction" does not exist / is not available.` but in remix.ethereum.org the screen goes blank as I click the debug button. What about this?

**Comment by Aniket-Engg** [comment]
> Regarding this issue, in remix-beta.ethereum.org it shows me the error that `The method "debug_traceTransaction" does not exist / is not available.` but in remix.ethereum.org the screen goes blank as I click the debug button. What about this?

To be specific, this thing will be fixed in v0.30.0 expected to be released today, and other issue will be targeted for v0.31.0

**Comment by m-waqas88** [comment]
> To be specific, this thing will be fixed in v0.30.0 expected to be released today, and other issue will be targeted for v0.31.0

After the release of  v0.30.0 I will again do the testing thoroughly and if resolved then will also close this issue.

Really appreciate the efforts of the remix team. Wish you guys success!

**Comment by m-waqas88** [comment]
>To be specific, this thing will be fixed in v0.30.0 expected to be released today, and other issue will be targeted for v0.31.0

@Aniket-Engg What about the release v0.30.0?

**Comment by Aniket-Engg** [comment]
@m-waqas88 v0.30.0 is live now.

**Comment by m-waqas88** [comment]
@Aniket-Engg thanks for informing timely. I am really happy with the new release. I have tested thoroughly and also tried to reproduce the same error. Now the screen is not going blank, instead when I start debugging, in the terminal it proactively gives me the message `The method "debug_traceTransaction" does not exist / is not available.`

I hope that issue #2529 will also be resolved in release v0.31.0
